File: stack/java/Stack.java
/** Class: Stack
Each instance represents a stack
*/
public class Stack {
   private int maxSize;
   private long[] stackArray;
   private int top;
   
   public MyStack(int maxSize) {
      this.maxSize = maxSize;
      stackArray = new long[maxSize];
      top = -1;
   }
  /**
    *Inserts an element on top of the stack
    *@param: element the element to be inserted.
    *@throws: Exception if the Stack overflows
    */
   public void push(long element) throws Exception{
      try{
        stackArray[++top] = element;
      }
     catch(java.lang.ArrayIndexOutOfBoundsException e){
       throw new Exception("Stack Overflow");
   }
  /**
    *@returns: Element at the top of the stack and removes the element from the stack
    *@throws: Exception if the Stack underflows
    */
   public long pop() throws Exception {
      try{
        return stackArray[top--];
      }
     catch(java.lang.ArrayIndexOutOfBoundsException e){
       throw new Exception("Stack Underflow");
   }
   }
  /**
    * @returns: element at top of the stack (view only) doesn't remove the element
    *@throws: Exception if the Stack underflows
    */
   public long peek() throws Exception {
      try{
        return stackArray[top];
      }
     catch(java.lang.ArrayIndexOutOfBoundsException e){
       throw new Exception("Stack Underflow");
   }
   }
  /**
    *@returns: true if the stack is empty
    */
   public boolean isEmpty() {
      return (top == -1);
   }
  
  /**
    *@returns: true if the stack is full
    */
   public boolean isFull() {
      return (top == maxSize - 1);
   }
}

File: stack/javascript/stack.js
// Stack class 
class Stack {

    // Implementation using Array
    constructor() {
        this.items = [];
    }

    // Related Functions

    // push function 
    push(element) {
        // inserts element into the items 
        this.items.push(element);
    }

    // pop function 
    pop() {
        // return top most element in the stack 
        // and removes it from the stack 
        // Underflow if stack is empty 
        if (this.items.length == 0)
            return "Underflow";
        return this.items.pop();
    }

    // peek function 
    peek() {
        // return the top most element from the stack 
        // but doesn't delete it. 
        return this.items[this.items.length - 1];
    }

    // isEmpty function 
    isEmpty() {
        // return true if stack is empty 
        return this.items.length == 0;
    }

    // printStack function 
    printStack() {
        var str = "";
        for (var i = 0; i < this.items.length; i++)
            str += this.items[i] + " ";
        return str;
    }
}


// Implementation
// creating object for stack class 
var stack = new Stack();

// returns true 
console.log(stack.isEmpty());

// returns Underflow 
console.log(stack.pop());

// Adding element to the stack 
stack.push(10);
stack.push(20);
stack.push(30);

// Printing the stack
console.log(stack.printStack());

// returns 30 
console.log(stack.peek());

// returns 30 and removes it from stack 
console.log(stack.pop());

console.log(stack.printStack());
